common-ts/src/utils/math.ts

```diff
@@ -0,0 +1,74 @@
+import { BN, L2OrderBook, PERCENTAGE_PRECISION } from '@drift-labs/sdk';
+
+const calculatemarkPrice = (bestBidPrice: BN, bestAskPrice: BN) => {
+	return bestBidPrice.add(bestAskPrice).divn(2);
+};
+
+const calculateBidAskAndmarkPrice = (l2: L2OrderBook) => {
+	const bestBidPrice = l2.bids.reduce((previousMax, currentBid) => {
+		if (!previousMax) return currentBid.price;
+		return BN.max(currentBid.price, previousMax);
+	}, undefined as BN);
+
+	const bestAskPrice = l2.asks.reduce((previousMin, currentBid) => {
+		if (!previousMin) return currentBid.price;
+		return BN.min(currentBid.price, previousMin);
+	}, undefined as BN);
+
+	const markPrice = calculatemarkPrice(bestBidPrice, bestAskPrice);
+
+	return {
+		bestBidPrice,
+		bestAskPrice,
+		markPrice,
+	};
+};
+
+const calculateSpreadQuote = (bestBidPrice: BN, bestAskPrice: BN) => {
+	return bestBidPrice.sub(bestAskPrice).abs();
```

Comment:
should have @0xbigz look at this, if they're crossing we might want it to show 0

Reply:
actually yea i think we showed negative before? confirm with z

Reply:
is showing negative bad? assuming now dlob construction/render issues, if theres actually that level of spread its represents makers subsidizing their maker rebate. also be nice to know if roundtrip exceeds taker fee in arbitrage cases